### ISSUE TITLE
Dashboard API doesn't handle non-existence of timeRange.{from,to}

### ISSUE THREAD

**Issue by GiedriusS** [grafana/grafana#23669]
Coming from https://github.com/grafana-tools/sdk/issues/83.

**What would you like to be added**:

I'm not a TS expert and I am not knowledgeable about the innards of Grafana but currently in the code we have:

```
public/app/features/templating/template_srv.ts:      const from = this.timeRange.from.valueOf().toString();
public/app/features/templating/template_srv.ts:      const to = this.timeRange.to.valueOf().toString();
```

It might happen when creating a dashboard via `POST /api/dashboards/db` that `timeRange` _will_ exist but {to,from} will not due to user error. The API should handle this invalid input, IMHO. Probably some sane values for `to` and `from` can be chosen if they were not provided.

**Why is this needed**:

So that users would be able to see their dashboards if the creation operation had succeeded according to the returned status code.

**Comment by dprokop** [comment]
@GiedriusS thank's for reporting the issue.

@marefr I'm not knowledgable about Grafana dashboard API but would it make sense to provide those default values when the API is used or you would rather like to handle this on the front-end?

**Comment by marefr** [comment]
Yeah would make sense to add some validation. Labeled as feature request.
